crates/interpreter/src/interpreter/shared_memory.rs

```diff
@@ -16,7 +16,7 @@ use std::{rc::Rc, vec::Vec};
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct SharedMemory {
     /// The underlying buffer.
-    buffer: Rc<RefCell<Vec<u8>>>,
+    buffer: Option<Rc<RefCell<Vec<u8>>>>,
```

Comment:
We can go without option, and just initialise with default empty vec.

Reply:
Rc allocates on its own